openstack/containerinfra/v1/clustertemplates/requests.go

```diff
@@ -13,29 +13,33 @@ type CreateOptsBuilder interface {
 
 // CreateOpts params
 type CreateOpts struct {
-	Labels              map[string]string `json:"labels,omitempty"`
-	FixedSubnet         string            `json:"fixed_subnet,omitempty"`
+	Name                string            `json:"name,omitempty"`
+	COE                 string            `json:"coe" required:"true"`
+	ImageID             string            `json:"image_id" required:"true"`
+	FlavorID            string            `json:"flavor_id,omitempty"`
 	MasterFlavorID      string            `json:"master_flavor_id,omitempty"`
-	NoProxy             string            `json:"no_proxy,omitempty"`
-	HTTPSProxy          string            `json:"https_proxy,omitempty"`
-	HTTPProxy           string            `json:"http_proxy,omitempty"`
-	TLSDisabled         bool              `json:"tls_disabled,omitempty"`
+	DNSNameServer       string            `json:"dns_nameserver,omitempty"`
 	KeyPairID           string            `json:"keypair_id,omitempty"`
-	Public              bool              `json:"public,omitempty"`
-	DockerVolumeSize    int               `json:"docker_volume_size"`
-	ServerType          string            `json:"server_type,omitempty"`
 	ExternalNetworkID   string            `json:"external_network_id,omitempty"`
-	ImageID             string            `json:"image_id" required:"true"`
+	FixedNetwork        string            `json:"fixed_network,omitempty"`
+	FixedSubnet         string            `json:"fixed_subnet,omitempty"`
+	NetworkDriver       string            `json:"network_driver,omitempty"`
+	APIServerPort       int               `json:"apiserver_port,omitempty"`
+	DockerVolumeSize    int               `json:"docker_volume_size,omitempty"`
```

Comment:
It's required.

Reply:
According to https://docs.openstack.org/magnum/latest/install/launch-instance.html, `DockerVolumeSize` isn't required. And in the JSON request I pasted yesterday, it's omitted.

Reply:
Unless you mean "required" as in "it must be included in the request body, even if it's null" ?

Reply:
These fields will work with omitempty.

Reply:
That's a change we may need to make in openstack client, for now, user can create cluster template without the docker_volume_size setting, but it will be failed when user try to create cluster with that template. I will double check that and get back.

Reply:
ahh understood.

So while it logically makes sense to require `docker_volume_size` because 99% of the time it's needed in order to create a cluster, that kind of validation is too high above the API for us to be worried about.

Gophercloud is a very thin, low communication layer between OpenStack and the Go language. If the API service will happily accept a cluster with no `docker_volume_size`, then we should allow that, even though the user will probably be shooting themselves in the foot. But at the same time, maybe there is such a possible template where `docker_volume_size` is not required.

Reply:
Another way of thinking about this is: always assume Gophercloud will _not_ be end-user facing. Gophercloud is meant for developers to use to create OpenStack compatible applications. It is in those applications where the enforcement of settings such as `docker_volume_size` will be required.

Reply:
That totally makes sense, cheers.